dacadeorg/dacade-frontend-app
# feat: implement ICP verfiable Credential with Dacade as relying party

# Submit a pull request

- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [x] The code style guideline have been followed.
- [x] Documentation has been updated to reflect your changes.
- [ ] Tests have been added or updated to reflect your changes.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details
This PR implements ICP verifiable Credential for ICP courses completion with Dacade as relying party.

## How to test
For now the test need to happen locally due module resolution incompatibity of the Verifiable Credential SDK and Next.js mendatory module resolution.

These are the step to follow locally:
1. Make sure to have node 22 and above
2. Run the app locally
3. Navigate to an ICP achievement
4. Click Request Credential button 

> Note: This demo runs on ICP non official identity provider and uses a testing credential issuer

More info:
- https://internetcomputer.org/docs/current/developer-docs/identity/verifiable-credentials/overview
- https://internetcomputer.org/docs/current/developer-docs/identity/verifiable-credentials/how-it-works#test-verifiable-credentials
